### ISSUE TITLE
Arbitrary camera intrinsic matrix support

### ISSUE THREAD

**Issue by mks0601** [intel-isl/Open3D#1417]
**Is your feature request related to a problem? Please describe.**
I'm trying to use `convert_from_pinhole_camera_parameters` function for the mesh visualization. Currently, it seems princpts of camera intrinsic matrix (`c_x` and `c_y`) should be image centers as reported in [[1](https://github.com/intel-isl/Open3D/issues/727)] and [[2](https://github.com/intel-isl/Open3D/issues/1164)].

**Describe the solution you'd like**
However, princpts don't have to be image centers, especially, when cameras are calibrated. Could you support arbitrary camera intrinsic matrix (i.e., arbitrary princpts) so that I can use calibrated princpts instead of image centers?

**Describe alternatives you've considered**
Maybe no alternatives.

**Additional context**
As reported in the issues, when I set `c_x` and `c_y` as calibrated ones (not image centers), I got an error message like below.
```
[Open3D WARNING] [ViewControl] ConvertFromPinholeCameraParameters() failed because window height and width do not match.
```

Thanks for sharing such a nice framework!

**Comment by saulzar** [comment]
Would second this request, I've been also struggling to understand this error message... my camera principle points are not at the image centre.

**Comment by Zumbalamambo** [comment]
+1

**Comment by theNded** [comment]
Resolved in #2564. Closed for now.
